include/aikido/statespace/Rn.hpp

```diff
@@ -9,33 +9,46 @@ namespace statespace {
 
 // Defined in detail/Rn-impl.hpp
 template <class>
-class RealVectorStateHandle;
+class RStateHandle;
 
-/// Represents a n-dimensional real vector space with vector addition as the
-/// group operation.
-class Rn : public virtual StateSpace
+/// Represents a N-dimensional real vector space with vector addition as the
+/// group operation. The dimension is unchangeable.
+template <int N>
+class R : public virtual StateSpace
 {
 public:
-  /// Point in a \c Rn.
+  /// Point in a \c R<N>.
   class State : public StateSpace::State
   {
+  public:
+    static constexpr int DimensionAtCompileTime = N;
+
   protected:
     State() = default;
     ~State() = default;
 
-    friend class Rn;
+    friend class R<N>;
   };
 
-  using StateHandle = RealVectorStateHandle<State>;
-  using StateHandleConst = RealVectorStateHandle<const State>;
+  /// Dimension of the space
+  static constexpr int DimensionAtCompileTime = N;
+
+  using Vectord = Eigen::Matrix<double, N, 1>;
+
+  using StateHandle = RStateHandle<State>;
+  using StateHandleConst = RStateHandle<const State>;
 
   using ScopedState = statespace::ScopedState<StateHandle>;
   using ScopedStateConst = statespace::ScopedState<StateHandleConst>;
 
-  /// Constructs a \c _dimension dimensional real vector space.
-  ///
-  /// \param _dimension dimension of the space
-  explicit Rn(int _dimension);
+  /// Constructs a real vector space. If N is Eigen::Dynamic (i.e., -1), this
+  /// state space becomes a dynamic size real vector space; if N is a
+  /// non-negative, then this state space becomes a \c N dimensional fixed size
+  /// real vector space.
+  R();
```

Comment:
This should throw an exception if `N` is `Eigen::Dynamic`, because there is no way to resize it.